src/js/webrequest.js

```diff
@@ -1770,11 +1777,25 @@ function dispatcher(request, sender, sendResponse) {
     // implications of accepting pbSurrogateMessage events
     // from third-party scripts in nested frames
     if (sender.frameId > 0) {
-      if (!request.frameUrl.startsWith('https://cdn.embedly.com/')) {
+      let frame_origin = sender.origin;
+
+      if (!utils.hasOwn(sender, "origin")) {
+        if (request.frameUrl) {
+          let path = (new URL(request.frameUrl)).pathname,
+            path_idx = request.frameUrl.indexOf(path);
+          frame_origin = request.frameUrl.slice(0, path_idx);
```

Comment:
Use `new URL(request.frameUrl)).origin` instead?

Reply:
Seems cleaner to me, if it works when tested on a browser without MessageSender.origin